torch/_linalg_utils.py

```diff
@@ -0,0 +1,108 @@
+"""Various linear algebra utility methods for internal use.
+
+"""
+
+import torch
+
+
+def is_sparse(A):
+    """Check if tensor A is a sparse tensor"""
+    if isinstance(A, torch.Tensor):
+        return A.layout == torch.sparse_coo
+    raise TypeError("expected Tensor but got %s" % (type(A).__name__))
+
+
+def get_floating_dtype(A):
+    """Return the floating point dtype of tensor A.
+
+    Integer types map to float32.
+    """
+    dtype = A.dtype
+    if dtype in (torch.float16, torch.float32, torch.float64):
+        return dtype
+    return torch.float32
+
+
+def matmul(A, B):
+    """Multiply two matrices.
+
+    If A is None, return B. A can be sparse or dense. B is always
+    dense.
+    """
+    # type: (Optional[Tensor], Tensor) -> Tensor
+    if A is None:
+        return B
+    if is_sparse(A):
+        return torch.sparse.mm(A, B)
+    return torch.matmul(A, B)
+
+
+def conjugate(A):
+    """Return conjugate of tensor A.
+
+    .. note:: If A's dtype is not complex, A is returned.
+    """
+    if A.dtype in [torch.complex32, torch.complex64, torch.complex128]:    
```

Comment:
`is_complex` implemented in https://github.com/pytorch/pytorch/pull/33175, so a simpler check can be done here

Reply:
Thanks, applied.